Enterprise-CMCS/managed-care-review
# MCR-3944: ReviewSubmit updated to use contract and rates

## Summary
**Primarily**

- Implemented V2 versions of the ReviewSubmit page and its sub components to consume graphql contract and rate types

**Bug fixes**

- Added attestation fields to the GQL contractFormData type
- Updated non unique GQL fragment names to resolve `Not all fragments have an unique name` on GQL generation
- Fixed `undefined fakesha` error on storybook for ContractDetails
- Updated the GQL `StateContact` type to use `titleRole` instead of `title` to match the other contact types like actuary
- Updated fetchContract to pass `updatedAt` and `createdAt` fields on the contract (included making updates to the related prisma functions and fetchContract resolver)

**Out of scope/not covered in this ticket**
- Navigating to the submission page from the Review Submit page
- Updating the unlock modal component to use contracts instead of HPP will be separated out into its own ticket
- Separate ticket to use `makeDocumentTableLookup` for ReviewSubmit will be created due to the function relying on a `revisionLookup` and HPP related types

**Remaining To Dos For this PR**

- [x] Fix server and unit test
- [x] Look into UI bug with contact detail's edit button
- [ ] Add test for ReviewSubmitV2

#### Related issues

https://jiraent.cms.gov/browse/MCR-3944

#### Screenshots
<img width="917" alt="Screenshot 2024-03-05 at 12 33 29 PM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/c0bb68b9-dd57-40d5-954a-297048aa2921">
<img width="556" alt="Screenshot 2024-03-05 at 12 33 38 PM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/5f687ef9-2f80-4d67-83b3-745cd0609cbc">
<img width="832" alt="Screenshot 2024-03-06 at 4 10 38 PM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/29738392-8382-4479-b792-54cf1f38c32b">

<img width="625" alt="Screenshot 2024-03-05 at 12 34 09 PM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/53e0e0fc-9d57-445b-86bd-3a2b6e2b783d">
<img width="577" alt="Screenshot 2024-03-05 at 12 34 00 PM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/1fd4d6cc-b444-4b10-976e-2fa97182e8c5">


#### Test cases covered

- With the exception of the ReviewSubmitV2 page, all V2 sub components (RateDetails, ContactSummary, ContractSummary, SubmissionTypeSummary) have had their tests updated to work with contracts and rates

## QA guidance

- Use the app as a state user and create a new submission. Should be able to reach and view the Review Submit page (actual submission is out of scope)
- Cypress test with linked rates off passes
- server and unit tests are passing
